CMakeLists.txt

```diff
@@ -68,11 +68,14 @@ file(WRITE ${CMAKE_BINARY_DIR}/coverage_targets.txt "")
 
 # Determine platform to build
 if(AZ_PLATFORM_IMPL STREQUAL "WIN32")
-  # build windows platform
   set(PAL az_win32)
 elseif(AZ_PLATFORM_IMPL STREQUAL "POSIX")
-  # build linux platform
   set(PAL az_posix)
+elseif(AZ_PLATFORM_IMPL STREQUAL "CUSTOM")
```

Comment:
What if, instead of making two options, you set  AZ_PLATFORM_IMPL to anything other than `win32` or `posix` (and, case-insensitive too), that becomes the custom platform lib name?
Other thing, I still think we should have `auto`, `auto` should be the default, and it becomes `win32` if target is windows (`IF(WIN32)`), and `posix` on POSIX (`if(UNIX)`). That should be good enough for common case, and if that doesn't work for you, you specify the platform, just as you do today.

Reply:
We can make it path to a source file, so if it is set, we would add that .c file to the `add_library()` list when compiling core.

Reply:
@antkmsft :
> We can make it path to a source file, so if it is set, we would add that .c file to the add_library() list when compiling core.

Why would it be better or more convenient than the typical approach (as in this PR) to have it as a library? What if the file requires special compilation options, special include folders or special defines? Please don't overcomplicate this, as different people have different needs and not all of them can be satisfied with the approach you propose.

On the other hand if someone would prefer to do what you say, he/she can do that WITHOUT any special support - either by adding more sources to the az_core library with `add_sources()` or by using various CMake trickery (like OBJECT library, or an empty INTERFACE library which specifies additional sources, or ...).

The approach of specifying a custom library is very versatile and universal, it has many benefits and no (at least in my opinion) drawbacks. Whether or not there will be `AZ_PLATFORM_IMPL` which has to be set to `CUSTOM` + a variable with target name or if `AZ_PLATFORM_IMPL` should be either `win32`, `posix` or `<name-of-the-target>` is purely cosmetic and doesn't make much of a difference.